### PR TITLE
fix: when query jenkins API only exception to throw error

### PR DESCRIPTION
**Pull request by zdtsw**
- either return data if 2XX
	- or return null for difference reasons

Fix: https://github.com/adoptium/ci-jenkins-pipelines/issues/556

### CODE REVIEW

**Comment by zdtsw** [comment]
run tests quick 19

**Comment by zdtsw** [comment]
run tests quick 19

File: pipelines/build/prTester/pr_test_pipeline.groovy [hunk 1]
```diff
@@ -64,14 +65,14 @@ class PullRequestTestPipeline implements Serializable {
         Map<String, ?> config = generateConfig(javaVersion)
         context.checkout([$class: 'GitSCM', userRemoteConfigs: [[url: config.GIT_URL]], branches: [[name: branch]]])
 
-        context.println "JDK${javaVersion} disableJob = ${config.disableJob}"
+        context.println "JDK${javaVersion} disableJob = ${config.disableJob}"
```
Comment (karianna) [inline pipelines/build/prTester/pr_test_pipeline.groovy:68]:
```suggestion
        context.println "JDK${javaVersion} disableJob = ${config.disableJob}"
```

File: pipelines/build/prTester/pr_test_pipeline.groovy [hunk 2]
```diff
@@ -64,14 +65,14 @@ class PullRequestTestPipeline implements Serializable {
...
-        context.println "JDK${javaVersion} disableJob = ${config.disableJob}"
+        context.println "JDK${javaVersion} disableJob = ${config.disableJob}" 
         context.jobDsl targets: DEFAULTS_JSON['templateDirectories']['upstream'], ignoreExisting: false, additionalParameters: config
     }
 
     /*
     * Main function, called from kick_off_tester.groovy by job "openjdk-build-pr-tester"
     */
-    def runTests() {
+    def runTests() {
```
Comment (karianna) [inline pipelines/build/prTester/pr_test_pipeline.groovy:75]:
```suggestion
    def runTests() {
```

**Review by karianna — APPROVED** [review]
lgtm

**Review by andrew-m-leonard — APPROVED** [review]
lgtm
This may prompt a CI admin approve scripting for HttpURLConnection

**Comment by andrew-m-leonard** [comment]
run tests

**Comment by zdtsw** [comment]
> lgtm This may prompt a CI admin approve scripting for HttpURLConnection

Yeah, now  I see the job failed with 
```
java.lang.Exception: [ERROR] Failure on jenkins api connection or parsing.
org.jenkinsci.plugins.scriptsecurity.sandbox.RejectedAccessException: Scripts not permitted to use staticField java.net.HttpURLConnection HTTP_OK
```
Could @karianna or @andrew-m-leonard approve this in jenkins?

**Comment by karianna** [comment]
approved

**Pull request merged**